services/movie.py

```diff
@@ -28,13 +33,16 @@
     genres_ids: list = None,
     actors_ids: list = None,
 ) -> Movie:
-    movie = Movie.objects.create(
+    movie = Movie(
         title=movie_title,
         description=movie_description,
     )
-    if genres_ids:
-        movie.genres.set(genres_ids)
-    if actors_ids:
-        movie.actors.set(actors_ids)
+
+    with transaction.atomic():
+        movie.save()
+        if genres_ids:
+            movie.genres.set(genres_ids)
+        if actors_ids:
+            movie.actors.set(actors_ids)
 
     return movie
```

Comment:
The `save()` method is called only once, which is good, but the code can be improved by moving the `save()` method call to the end of the `with transaction.atomic():` block. This way, if anything fails while setting the genres and actors, the movie won't be saved, which is a good practice when dealing with transactions.